### ISSUE TITLE
CSV without header

### ISSUE THREAD

**Issue by rabbit72** [di/vladiate#88]
Is it possible to validate files without header?

I expected something like this but it seems it's not possible.

`validators = {
        0: [
            UniqueValidator(),
            IntValidator(),
        ],
        1: [
            FloatValidator(),
        ],
        2: [
            SetValidator(["USD", "EUR", "GBP"]),
        ],
    }`

**Comment by di** [comment]
Not currently, the header is required for certain validators to function properly, e.g. to determine row lengths for the `RowLengthValidator`, as well as output on results.

**Comment by jonafato** [comment]
@rabbit72 I put together a small pull request at #93 as a proof of concept. This would allow a developer to define the header values at the validator level to validate CSV files without included header rows. Would this change address your use case?

**Comment by rabbit72** [comment]
@jonafato hi, looks promising!
Does it mean this change allows me to name columns in a CSV file that doesn’t contain the first row with columns?
Do you validate that all rows contain the same number of columns?

**Comment by jonafato** [comment]
> Does it mean this change allows me to name columns in a CSV file that doesn’t contain the first row with columns?

Yes, if you use this feature, your CSV files should not include header rows. (This feature is not compatible with CSV files that include header rows, and setting `fieldnames` along with a header row is likely to result in a validation error or other invalid results.)

> Do you validate that all rows contain the same number of columns?

This is not enabled by default, but `vladiate` ships with a [`RowLengthValidator`](https://github.com/di/vladiate?tab=readme-ov-file#built-in-validators) that you can use with the [`row_validators`](https://github.com/di/vladiate?tab=readme-ov-file#running-vlads-programatically) attribute to ensure that all rows have the same number of columns as the header row (or, with this new feature PR, the `fieldnames` attribute). This row validator is available for use as of version 0.0.25.
